pdftext/pdf/chars.py

```diff
@@ -19,11 +19,20 @@ def update_previous_fonts(char_infos: List, i: int, prev_fontname: str, prev_fon
         char_infos[j]["font"]["flags"] = fontflags
 
 
-def get_pdfium_chars(pdf, page_range, fontname_sample_freq=settings.FONTNAME_SAMPLE_FREQ):
+def get_pdfium_chars(pdf, page_range, flatten_pdf, fontname_sample_freq=settings.FONTNAME_SAMPLE_FREQ):
     blocks = []
 
     for page_idx in page_range:
         page = pdf.get_page(page_idx)
+
+        if flatten_pdf:
+            # Flatten form fields and annotations into page contents.
+            page._flatten(flag=pdfium_c.FLAT_NORMALDISPLAY)
```

Comment:
(Note that this will change to just `flatten` (without underscore) once v5 is merged.)

Reply:
Great @mara004 . I'll keep an eye out to change it. Thanks

Reply:
It might make more sense to simply copy the function's code into this project to avoid any version trouble. It's quite an easy wrapper:
```python
def flatten(page, flag=pdfium_c.FLAT_NORMALDISPLAY):
    if not self.formenv:
        raise RuntimeError("flatten(page) requires previous pdf.init_forms() before page retrieval.")
    rc = pdfium_c.FPDFPage_Flatten(page, flag)
    if rc == pdfium_c.FLATTEN_FAIL:
        raise PdfiumError("Failed to flatten annotations / form fields.")
    return rc
```
Or maybe just inline the `FPDFPage_Flatten()` call and return code check.